how-to/integrate-with-sso/integrate-with-auth0/client/src/auth.ts

```diff
@@ -259,31 +293,50 @@ async function authenticatedStateChanged(
 	pollTimerStart(isAuthenticated);
 }
 
-function pollTimerStop() {
+/**
+ * Stop the token verification polling timer.
+ */
+function pollTimerStop(): void {
 	if (pollTimerId) {
 		window.clearInterval(pollTimerId);
 		pollTimerId = undefined;
 	}
 }
 
-function pollTimerStart(isAuthenticated: boolean) {
-	if ((authSettings.verifyPollMs ?? 0) > 0 && isAuthenticated) {
+/**
+ * Start the token verification polling timer.
+ * @param isAuthenticated Is the user authenticated.
+ */
+function pollTimerStart(isAuthenticated: boolean): void {
+	if ((auth0Settings.verifyPollMs ?? 0) > 0 && isAuthenticated) {
 		pollTimerId = window.setInterval(async () => {
 			const accessToken = loadProperty(STORE_ACCESS_TOKEN);
-			const userInfo = await checkTokenValidity(accessToken, false);
-			if (!userInfo) {
-				informationCallback("Access token no longer valid, logout");
-				removeProperty(STORE_ACCESS_TOKEN);
+			if (accessToken) {
+				const userInfo = await checkTokenValidity(accessToken, false);
+				if (!userInfo) {
+					informationCallback("Access token no longer valid, logout");
+					removeProperty(STORE_ACCESS_TOKEN);
+					removeProperty(STORE_AUTH_STATE);
+					await authenticatedStateChanged(false);
+				} else {
+					informationCallback("Access token still valid");
+				}
+			} else {
+				informationCallback("Access token not found, logout");
 				removeProperty(STORE_AUTH_STATE);
 				await authenticatedStateChanged(false);
-			} else {
-				informationCallback("Access token still valid");
 			}
-		}, authSettings.verifyPollMs);
+		}, auth0Settings.verifyPollMs);
 	}
 }
 
-async function checkForUrls(win: OpenFin.Window, urls: string[]): Promise<URL | undefined> {
+/**
+ * Check to see if the window is displaying one of the specified urls with a partial match.
+ * @param win The window to check.
+ * @param urls The urls to look for.
+ * @returns The complete url if it has one it was looking for.
+ */
+async function checkForUrls(win: OpenFin.Window | undefined, urls: string[]): Promise<URL | undefined> {
```

Comment:
Could we use the new navigation event in https://developer.openfin.co/versions/?product=Runtime#/?product=Runtime&version=30.110.74.8 to get rid of polling to see if the url changed?

Reply:
Updated auth window to use the `url-changed` event.